File: dart/com/spoonacular/client/model/inline_response20045.dart
part of openapi.api;

class InlineResponse20045 {
  
  List<String> pairedWines = [];
  
  String pairingText = null;
  
  List<InlineResponse20045ProductMatches> productMatches = [];
  InlineResponse20045();

  @override
  String toString() {
    return 'InlineResponse20045[pairedWines=$pairedWines, pairingText=$pairingText, productMatches=$productMatches, ]';
  }

  InlineResponse20045.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['pairedWines'] == null) {
      pairedWines = null;
    } else {
      pairedWines = (json['pairedWines'] as List).cast<String>();
    }
    if (json['pairingText'] == null) {
      pairingText = null;
    } else {
          pairingText = json['pairingText'];
    }
    if (json['productMatches'] == null) {
      productMatches = null;
    } else {
      productMatches = InlineResponse20045ProductMatches.listFromJson(json['productMatches']);
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (pairedWines != null)
      json['pairedWines'] = pairedWines;
    if (pairingText != null)
      json['pairingText'] = pairingText;
    if (productMatches != null)
      json['productMatches'] = productMatches;
    return json;
  }

  static List<InlineResponse20045> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20045>() : json.map((value) => new InlineResponse20045.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20045> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20045>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20045.fromJson(value));
    }
    return map;
  }
}

File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20045.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20045ProductMatches
import scala.collection.immutable.Seq

/**
 * 
 * @param pairedWines 
 * @param pairingText 
 * @param productMatches 
 */
case class InlineResponse20045(pairedWines: Seq[String],
                pairingText: String,
                productMatches: Seq[InlineResponse20045ProductMatches]
                )

object InlineResponse20045 {
    /**
     * Creates the codec for converting InlineResponse20045 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20045] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20045] = deriveEncoder
}
